app/models/trip.rb

```diff
@@ -0,0 +1,15 @@
+class Trip < ApplicationRecord
+  belongs_to :driver
+  belongs_to :passenger
+  validates :rating, presence: true, format: { with: /^[1-5]$/, multiline: true} #multiline to allow the start karat and end dollar sign in regex
+
+  #task is independent of context -- biz logic only
+  def earnings
```

Comment:
I'm not sure what this method is for.  You're calculating the total earnings for all the trips, regardless of driver.

Reply:
It works on our Driver list page where they are shown their total earnings.